### DISCUSSION TITLE
Getting a custom string from DatePicker

### DISCUSSION THREAD

**Discussion by TransGirlCodes** [Appsilon/shiny.fluent#180] [Q&A]
Hi, thanks for shiny.fluent!

Is it possible to get a custom string for the `DatePicker.shinyInput`?

I read the docs for the package, and Microsoft's React docs and found the format formatDate.

So I was able do do:

```R
Date.Picker.shinyInput(
  ...,
  formatDate = JS("(date) => {return !date ? '' : date.getFullYear() + '.' + (date.getMonth() + 1);}")
)
```

But that only alters how the date is displayed in the ui component (YYYY.MM), I still get a full date-time string back e.g. `2023-02-28T13:18:13.400Z`

Is it possible to have the component only return "YYYY.MM" as the `input$*` value, or am I forced to do string manip on the R side?

**Comment by jakubsob** [comment]
Hi @SabrinaJaye,

currently this is not a supported feature and value accessible in the server will be a full date-time string. 

We will investigate possibility of adding such parsing of input value to this component, but until that happens I suggest parsing the date on R side.
